### PR TITLE
Sound fix on silent mode

### PR DESCRIPTION
**Pull request by zacharya19**
Don't say I'm not spoiling you

### CODE REVIEW

File: app/src/main/java/com/sklein/ubimo_pager/MainActivity.java [hunk 1]
```diff
@@ -1,32 +1,44 @@
...
 import android.view.View;
+import android.view.WindowManager;
 import android.widget.ImageButton;
 import android.widget.TextView;
 
 public class MainActivity extends AppCompatActivity {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
+        getWindow().addFlags(WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED);
```
Comment (shmulik-klein) [inline app/src/main/java/com/sklein/ubimo_pager/MainActivity.java]: Can you please put it after the call for the super method ?

File: app/src/main/java/com/sklein/ubimo_pager/MainActivity.java [hunk 2]
```diff
@@ -1,32 +1,44 @@
...
         super.onCreate(savedInstanceState);
         Intent smsIntent = getIntent();
         Bundle smsBundle = smsIntent.getExtras();
         if (null != smsBundle) {
             setContentView(R.layout.activity_main);
 
+            final AudioManager audio = (AudioManager) getSystemService(Context.AUDIO_SERVICE);
+            final int music = AudioManager.STREAM_MUSIC;
```
Comment (shmulik-klein) [inline app/src/main/java/com/sklein/ubimo_pager/MainActivity.java]: * You can use `AudioManager.STREM_MUSIC` enum directly in `setStreamVolume()` call, it's a little more verbose but also more readable.
* Save the current volume value with `getStreamVolume()` before setting it on maximum
* Revert the volume to original on `onClickListener()`

File: app/src/main/java/com/sklein/ubimo_pager/WelcomeActivity.java [hunk 3]
```diff
@@ -24,15 +29,21 @@ public void onClick(View v) {
...
+    private void checkForPermission(String[] permissions) {
+        ArrayList<String> permissionsToAsk = new ArrayList<String>();
+        for (String permission : permissions) {
+            int permissionStatus = ContextCompat.checkSelfPermission(this, permission);
+            if (permissionStatus != PackageManager.PERMISSION_GRANTED) {
+                permissionsToAsk.add(permission);
+            }
+        }
+
+        if (!permissionsToAsk.isEmpty()) {
```
Comment (shmulik-klein) [inline app/src/main/java/com/sklein/ubimo_pager/WelcomeActivity.java:45]: Why do you want to ask twice for the permissions ?

**Review by shmulik-klein — CHANGES_REQUESTED** [review]
Good work you FUCKER!

File: app/src/main/java/com/sklein/ubimo_pager/WelcomeActivity.java (hunk 3 above)
Comment (zacharya19) [reply app/src/main/java/com/sklein/ubimo_pager/WelcomeActivity.java:45]: I'm actually asking them all together

Comment (shmulik-klein) [reply app/src/main/java/com/sklein/ubimo_pager/WelcomeActivity.java:45]: Right! my mistake, I thought I saw `requestPermissions` usage in the first loop

File: app/src/main/java/com/sklein/ubimo_pager/WelcomeActivity.java [hunk 4]
```diff
@@ -24,15 +29,21 @@ public void onClick(View v) {
...
-        checkForPermission(Manifest.permission.READ_SMS);
+        checkForPermission(NEEDED_PERMISSIONS);
     }
 
-    private void checkForPermission(String permission) {
-        final int permissionStatus = ContextCompat.checkSelfPermission(this, permission);
-        if (permissionStatus == PackageManager.PERMISSION_GRANTED) {
-            return;
+    private void checkForPermission(String[] permissions) {
+        ArrayList<String> permissionsToAsk = new ArrayList<String>();
```
Comment (shmulik-klein) [inline app/src/main/java/com/sklein/ubimo_pager/WelcomeActivity.java]: Use interfaces whenever you can - `List<String> permissionsToAsk = new ArrayList<String>();`

**Pull request merged**